### PR TITLE
desed: init at 1.2.1

### PR DESCRIPTION
**Pull request by vinylen**
## Description of changes

Debugger for Sed: demystify and debug your sed scripts, from comfort of your terminal. 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by vinylen** [comment]
@NixOS/nixpkgs-merge-bot merge

File: pkgs/by-name/de/desed/package.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  src = fetchFromGitHub {
+    owner = "SoptikHa2";
+    repo = "desed";
+    rev = "v${version}";
+    hash = "sha256-vmv9VAfhScVY6Kt+jblCebCMJ+3yCojgK3mLszjembY=";
+  };
+
+  cargoPatches = [
+    ./cargo-lock.patch
+  ];
```
[the remark this answers is not in the document]
Comment (vinylen) [reply pkgs/by-name/de/desed/package.nix]: Should be resolved now, thanks.

[the remark this answers is not in the document]
Comment (vinylen) [reply pkgs/by-name/de/desed/package.nix]: There is a standing issue in the project [here](https://github.com/SoptikHa2/desed/issues/27) to include the Cargo.lock file. That issue was created over a year ago and the author of the project has not responded.

**Comment by a-kenji** [comment]
Since https://github.com/SoptikHa2/desed/pull/29 is merged I recommend bumping to the latest unstable.

**Comment by vinylen** [comment]
> Since [SoptikHa2/desed#29](https://github.com/SoptikHa2/desed/pull/29) is merged I recommend bumping to the latest unstable. This allows us to drop the vendored lock file.

Nice! Does this look ok? https://github.com/NixOS/nixpkgs/compare/3aaf20d1803f6ff04d17971a2c65baf311dd2612..110e5ec788e56608f9aa35f1df4f539c09808985

**Comment by vinylen** [comment]
Maybe I should change the commit message? Since technically we're not init at 1.2.1 anymore?

**Comment by vinylen** [comment]
> @vinylen just rebase and rename it.

Done!

**Review by a-kenji — APPROVED** [review]
Thank you! One more thing though:
The version should now be `[latest-version]-unstable-[date-of-last-commit]`,
if that is fixed, I think this pr-is ready

**Comment by vinylen** [comment]
> Thank you! One more thing though: The version should now be `[latest-version]-unstable-[date-of-last-commit]`, if that is fixed, I think this pr-is ready

Updated!

**Pull request merged**